### ISSUE TITLE
Runtime Error when multipling complex32 with float32

### ISSUE THREAD

**Issue by quancs** [pytorch/pytorch#69209]
## 🐛 Bug

```
import torch
torch.tensor(1j, dtype=torch.complex32) * torch.tensor(1, dtype=torch.float32)
```

Runtime Error:
```
RuntimeError: common_dtype_ != ScalarType::Undefined INTERNAL ASSERT FAILED at "/opt/conda/conda-bld/pytorch_1616554793803/work/aten/src/ATen/TensorIterator.cpp":239, please report a bug to PyTorch.
```

## To Reproduce

1.
1.
1.

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

 - PyTorch Version (e.g., 1.0): 1.10
 - OS (e.g., Linux):
 - How you installed PyTorch (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

## Additional context

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @anjali411 @dylanbespalko @mruberry @Lezcano @nikitaved

**Comment by malfet** [comment]
`torch.complex32` is complex number based on half precision float. Half-precision complex support is very limited to say the least, although it would be nice to print a more sensible error (or upcast half to float)
Command you probably wanted to invoke is this
```
$ python3 -c "import torch;print(torch.tensor(1j, dtype=torch.complex64) * torch.tensor(1, dtype=torch.float32))"
tensor(0.+1.j)
```
 Tentatively grabbing for myself.

**Comment by quancs** [comment]
In my case, my code runs OK for 32 bit precision, but reported this error when I run my code for half precision training.
This error is solved for my code by not converting half precision float to half precision complex before multiplying.
I guess automaticly upcast to complex32 to complex64 is a good way.

**Comment by mruberry** [comment]
cc @anjali411 another reason to remove the ability to construct complex32 tensors

**Comment by anjali411** [comment]
@mruberry yup makes sense! I'll create a PR to do that soon since more of these issues have started surfacing lately.

**Comment by malfet** [comment]
Removing 1.11 milestone as `torch.complex32` is not available in 1.11

**Comment by kshitij12345** [comment]
Experimental support for `complex32` was added in 1.12 (which includes support for `multiply`). 

With that release,
```python
>>> import torch
>>> torch.tensor(1j, dtype=torch.complex32) * torch.tensor(1, dtype=torch.float32)
<stdin>:1: UserWarning: ComplexHalf support is experimental and many operators don't support it yet. (Triggered internally at aten/src/ATen/EmptyTensor.cpp:31.)
tensor(0.+1.j)
```
